omod/src/main/java/org/openmrs/module/ugandaemrreports/fragment/controller/SendReportRequestFragmentController.java

```diff
@@ -0,0 +1,188 @@
+import java.security.SecureRandom;
+import java.security.Security;
+import java.security.cert.CertificateException;
+import java.security.cert.X509Certificate;
+import java.util.Base64;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ *
+ */
+public class SendReportRequestFragmentController {
+
+    protected Log log = LogFactory.getLog(getClass());
+
+
+    public SimpleObject post(@SpringBean ReportService reportService,
+                             @RequestParam("request") String requestUuid) throws IOException {
+        ReportRequest req = reportService.getReportRequestByUuid(requestUuid);
+        if (req == null) {
+            throw new IllegalArgumentException("ReportRequest not found");
+        }
+
+        RenderingMode renderingMode = req.getRenderingMode();
+        String linkUrl = "/module/reporting/reports/reportHistoryOpen";
+
+        if (renderingMode.getRenderer() instanceof WebReportRenderer) {
+
+            throw new IllegalStateException("Web Renderers not yet implemented");
+        } else {
+            String filename = renderingMode.getRenderer().getFilename(req).replace(" ", "_");
+            String contentType = renderingMode.getRenderer().getRenderedContentType(req);
+            byte[] data = reportService.loadRenderedOutput(req);
+
+            if (data == null) {
+                throw new IllegalStateException("Error retrieving the report");
+            } else {
+
+                return sendData(data);
+
+            }
+
+        }
+    }
+
+    public SimpleObject previewReport(@SpringBean ReportService reportService,
+                             @RequestParam("request") String requestUuid) throws IOException {
+        ReportRequest req = reportService.getReportRequestByUuid(requestUuid);
+        if (req == null) {
+            throw new IllegalArgumentException("ReportRequest not found");
+        }
+        byte[] data=null;
+        RenderingMode renderingMode = req.getRenderingMode();
+        String linkUrl = "/module/reporting/reports/reportHistoryOpen";
+
+        if (renderingMode.getRenderer() instanceof WebReportRenderer) {
+
+            throw new IllegalStateException("Web Renderers not yet implemented");
+        } else {
+            String filename = renderingMode.getRenderer().getFilename(req).replace(" ", "_");
+            String contentType = renderingMode.getRenderer().getRenderedContentType(req);
+            data = reportService.loadRenderedOutput(req);
+        }
+        String jsonData=new String(data).replace("x-","x");
+
+       return SimpleObject.create("data",jsonData);
+    }
+
+
+
+    private SimpleObject sendData(byte[] data) throws IOException {
+        Map map = new HashMap();
+        SimpleObject simpleObject=new SimpleObject();
+        int responseCode = 0;
+        try {
+            URL url = new URL("https://ugisl.mets.or.ug:5000/ehmis");
+
+            String encoding = Base64.getEncoder().encodeToString(("mets.mkaye:METS4321!").getBytes("UTF-8"));
```

Comment:
@solemabrothers Do not hardcode the username and password to the HIM

Reply:
In your opinion what’s the best way to pass those credentials to the connection object.

Reply:
Removed